### PR TITLE
Screening refactoring

### PR DESCRIPTION
**Pull request by agusiol**
To create Screening with association I have to pass in controller all params separately(like params[:date] etc.) otherwise the IDs of associated  models are not passed to the create use case. Should it be like that or am I doing something wrong?

### CODE REVIEW

File: app/controllers/screenings_controller.rb [hunk 1]
```diff
@@ -14,7 +14,9 @@ def create
                   date: params[:screening][:date], 
                   movie_id: params[:movie_id],
                   cinema_hall_id: params[:cinema_hall_id] })
-    
+      #doesn't work
```
Comment (baniak66) [inline app/controllers/screenings_controller.rb:17]: Maybe place `byebug` inside of this method and check how your params look like :)

Comment (baniak66) [reply app/controllers/screenings_controller.rb:17]: Look also on the `screening_params` method and what params are permited.

Comment (agusiol) [reply app/controllers/screenings_controller.rb:17]: I added `:movie_id `and `:cinema_hall_id` but it didn't change anything 
```
   def screening_params
      params.require(:screening).permit(:date, :movie_id, :cinema_hall_id)
    end
```
   When I run byebug and check `screening_params` there are not there. So I guess I have to use `params` separately like I did?
   
montecinema — fsevent_watch ◂ localhost:3000)  montecinema  RBENV_VERSION=3 0 1 TERM_PROGRAM=Apple_Terminal — 126×47 2021-05-28 13-59-14

Comment (baniak66) [reply app/controllers/screenings_controller.rb:17]: Your `screening` parameters should all be under the `screening` key. If we take the above screenshot as an example it should look like this:
```
{ "screening" => { "date" => "2021-05-28", "cinema_hall_id" => "1", "movie_id" => "12" }, "controller" => "screening", "action" => "create" }
```

Comment (agusiol) [reply app/controllers/screenings_controller.rb:17]: Ok. Any idea why there are not? Where can I start looking for a bug?

**Pull request merged**